packages/itwinui-react/src/core/Popover/Popover.tsx

```diff
@@ -229,7 +229,11 @@ export const usePopover = (options: PopoverOptions & PopoverInternalProps) => {
     useHover(floating.context, {
       enabled: !!mergedInteractions.hover,
       delay: 100,
-      handleClose: safePolygon({ buffer: 1, requireIntent: false }),
+      handleClose: safePolygon({
+        buffer: 1,
+        requireIntent: false,
```

Comment:
Should we remove `requireIntent` now? I believe the main reason for making it `false` was because it was hard to tell if `safePolygon` was working at all.

Reply:
I believe it was intentional when working on #1942 since we didn't want to take mouse speed in account at that time.

But now, yes, I think it's better to remove it to make it easier for users who want to actually hover over the adjacent item instead of keeping the submenu open because of the safePolygon. Removed (https://github.com/iTwin/iTwinUI/pull/2026/commits/01626db0c6e24cf0d0ed02957f9de62820988be2).

Reply:
Hmm, now I'm noticing some really weird behaviors (at least in Firefox). Screen recording shows that it doesn't work 2/4 times, and worse, it completely breaks the hover trigger after it fails. Maybe that's why we had `requireIntent: false`?

https://github.com/iTwin/iTwinUI/assets/9084735/a8f6332c-3818-402c-a59e-80b93b3b0fd8

Reply:
After a bit of investigation, I found a few issues.

> and worse, it completely breaks the hover trigger after it fails

Looks like this was a small bug. Fixed in [`8a65e07`](https://github.com/iTwin/iTwinUI/pull/2026/commits/8a65e07b419472c18f2ea85f21285a85af611ef2).

> it doesn't work 2/4 times

Not sure of the exact reasoning of why that's happening. But I think it's because of some race conditions in how `hasFocusedNodeInSubmenu` is being set and used.

So, using something something like `queueMicrotask` or `setTimeout(() => { … }, 0)` keeps the submenus open more reliably but could be considered hacky (https://github.com/iTwin/iTwinUI/pull/2026/commits/405bfec71f9d22cee8a74f5eb6c86ab5175a4079). Open to alternatives.

After those commits:

<details>
<summary>Firefox</summary>

https://github.com/iTwin/iTwinUI/assets/45748283/e3904df6-8148-4b9f-b535-21b2f133a410

</details>

<details>
<summary>Edge</summary>

https://github.com/iTwin/iTwinUI/assets/45748283/eac5e561-1aa5-4a97-9a4e-14cdd06792ae

</details>